kubectl-rook-ceph.sh

```diff
@@ -226,14 +227,221 @@ function path_cm_rook_ceph_operator_config() {
 }
 
 ####################################################################################################
-# 'kubectl rook-ceph mon-endpoints' commands
+# 'kubectl rook-ceph mons' commands
 ####################################################################################################
 
+function run_mons_command () {
+  if [ "$#" -ge 1 ] && [ "$1" = "restore-quorum" ]; then
+    shift # remove the subcommand from the front of the arg list
+    run_restore_quorum "$@"
+  else
+    fetch_mon_endpoints "$@"
+  fi
+}
+
 function fetch_mon_endpoints() {
   end_of_command_parsing "$@" # end of command tree
   KUBECTL_NS_CLUSTER get cm rook-ceph-mon-endpoints -o json | jq --monochrome-output '.data.data' | tr -d '"' | tr -d '=' | sed 's/[A-Za-z]*//g'
 }
 
+wait_for_pod_of_deployment_to_be_running() {
+  deployment=$1
+  info_msg "Waiting for the pod from deployment \"$deployment\" to be running"
+  KUBECTL_NS_CLUSTER wait deployment $deployment --for condition=Available=True --timeout=90s
+}
+
+function run_restore_quorum() {
+  parse_flags parse_image_flag "$@" # parse flags before the good mon name
+  [[ -z "${REMAINING_ARGS[0]:-""}" ]] && fail_error "Missing healthy mon name"
+  good_mon="${REMAINING_ARGS[0]}"              # get the good mon being used to restore quorum
+  shift # remove the healthy mon from the front of the arg list
+  REMAINING_ARGS=("${REMAINING_ARGS[@]:1}")           # remove mon name from remaining args
+  end_of_command_parsing "$@" # end of command tree
+
+  # Parse the endpoints configmap for the mon endpoints
+  bad_mons=()
+  mon_endpoints=$(KUBECTL_NS_CLUSTER get cm rook-ceph-mon-endpoints -o jsonpath='{.data.data}')
+  # split the endpoints into an array, separated by the comma
+  for single_mon in ${mon_endpoints//,/ } ; do
+    mon_name=$(echo "${single_mon/=/ }" | awk '{print $1}')
+    mon_endpoint=$(echo "${single_mon/=/ }" | awk '{print $2}')
+    echo "mon=$mon_name, endpoint=$mon_endpoint"
+    if [ "$mon_name" = "$good_mon" ]; then
+      good_mon_public_ip=$(echo "${mon_endpoint/:/ }" | awk '{print $1}')
+      good_mon_port=$(echo "${mon_endpoint/:/ }" | awk '{print $2}')
+    else
+      bad_mons+=("$mon_name")
+    fi
+  done
+
+  # Parse the cluster FSID
+  ceph_fsid=$(KUBECTL_NS_CLUSTER get secret rook-ceph-mon -o jsonpath='{.data.fsid}' | base64 -d)
+  if [ -z ${good_mon_public_ip+x} ]; then
+    error_msg "error: good mon $good_mon not found"
+    exit 1
+  fi
+  if [ "$ceph_fsid" = "" ]; then
+    error_msg "ceph cluster fsid not found"
+    exit 1
+  fi
+
+  # Check for the existence of the toolbox
+  info_msg "Start the toolbox if it is not yet running"
+  wait_for_pod_of_deployment_to_be_running rook-ceph-tools
```

Comment:
maybe if we just say
`wait_for_pod_to_run`

Reply:
`rook-ceph-tools` is the name of a deployment, not a pod. How about `wait_for_deployment_to_run`?

Reply:
Looks good